### PR TITLE
add onError event and wire image load failure to it

### PR DESCRIPTION
**Pull request by mrousal**

### CODE REVIEW

File: ReactWindows/ReactNative.Shared/Views/Image/ReactImageLoadEvent.cs [hunk 1]
```diff
@@ -57,6 +63,18 @@ public ReactImageLoadEvent(int viewId, int eventType, string imageUri, int width
             _height = height;
         }
 
+		/// <summary>
```
Comment (reseul) [inline ReactWindows/ReactNative.Shared/Views/Image/ReactImageLoadEvent.cs]: Small alignment

File: ReactWindows/ReactNative.Shared/Views/Image/ReactImageLoadEvent.cs [hunk 2]
```diff
@@ -72,6 +90,8 @@ public override string EventName
                         return "topLoad";
                     case OnLoadEnd:
```
Comment (reseul) [inline ReactWindows/ReactNative.Shared/Views/Image/ReactImageLoadEvent.cs]: What do we do with this one, then? Doesn't seem to be used.
Semantically "onLoad" marks the end, anyway.

[edit]: Based on the comments of the JS counterpart, onLoadEnd should be called in both cases (error & success).
So order for success:
- onLoadstart, onLoad, onLoadEnd
for failure:
- onLoadStart, onError, onLoadEnd 

(not sure about the order of onLoadEnd vs onXXX, perhaps peeking into Android code would inspire somehow.)

File: ReactWindows/ReactNative.Shared/Views/Image/ReactImageLoadEvent.cs [hunk 3]
```diff
@@ -135,6 +155,12 @@ public override void Dispatch(RCTEventEmitter eventEmitter)
                 }
             }
 
+            if (_eventType == OnError)
+            {
+                eventData = new JObject();
+                eventData.Add("error", _error);
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/Views/Image/ReactImageLoadEvent.cs]: Use object initializer:
```
eventData = new JObject
{
    { "error", _error }
};
```

**Comment by rozele** [comment]
You should consider sending a similar PR to send the error message on Android.

**Comment by reseul** [comment]
@rozele Actually they dispatch from here: https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageView.java

And yes, it's first onload/onerror, THEN onLoadEnd.

File: ReactWindows/ReactNative.Shared/Views/Image/ReactImageLoadEvent.cs (hunk 2 above)
Comment (mrousal) [reply ReactWindows/ReactNative.Shared/Views/Image/ReactImageLoadEvent.cs]: updated to fire both events as on android

**Pull request merged**